docs/technical-overview.md

```diff
@@ -0,0 +1,241 @@
+
+Project source code is available in the following repositories:
+- https://github.com/thehyve/podium-registry — configuration registry and system health monitoring webapp;
+- https://github.com/thehyve/podium — core functionality.
+
+Project codebase is maintained by The Hyve.
+For technical questions you can contact alexey@thehyve.nl or gijs@thehyve.nl
+
+
+## Purpose of the document
+
+This document is intended to be a succinct technical reference that facilitates updating, testing and further development of the Podium project, and is aimed at people who plan to contribute to the project codebase.
+High-level architecture overview, component interactions and major dependencies are included, while implementation details that can be trivially derived from the code (e.g. DTOs) and/or autogenerated (e.g. Swagger API definition, database schema), and user-oriented nuances (e.g. visual elements of the web UIs, installation manual) are omitted.
+
+
+## Principal components
+
+### Overview
+
+
+```plantuml
+@startuml
+node "Registry" {
+  [Registry API]
+  [Admin Webapp] --> [Registry API]
+}
+
+node "UAA" {
+  [User Auth API] --> [Registry API]
+}
+
+node "Gateway" {
+  [Gateway API] --> [Registry API]
+  [Podium Webapp] --> [Gateway API]
+}
+
+cloud {
+  [Github Config]
+  [Registry API] --> [Github Config]
+}
+
+rectangle Common [
+  podium-common
+  * Shared DTOs
+  * Request type and status enums
+  * Event definitions
+  * Exception classes
+  * etc
+]
+
+[Gateway API] ..> Common
+[User Auth API] ..> Common
+
+
+database "PostgreSQL" {
+  frame "Auth DB" as AuthDB {
+  }
+  frame "Request DB" as RequestDB {
+  }
+}
+
+rectangle "Auth ES Index" as AuthES
+AuthES -[hidden]-> AuthDB
+rectangle "Request ES Index" as RequestES
+RequestES -[hidden]-> RequestDB
+
+@enduml
+```
+
+
+### Registry
+
+The registry is a system administration component that plays a role of a service discovery hub and a configuration provider used by the other components of the Podium system.
+
+Registry webapp displays the system health status, lists running component instances, shows configuration options & environment variables, and gives access to various metrics that might be useful to monitor the system status.
+The webapp fetches all data from the registry backend only, it does not interact with any other components.
+Since the webapp is not supposed to be accessible to end users and is not essential to run the Podium system; it is reasonable to assume that non-backward compatible changes are completely acceptable during the upgrade as long as a technical user can find metrics necessary for operations & maintenance.
+
+Supported NodeJS version: 8.12.0+; supported AngularJS version: 1.7.5.
```

Comment:
I wouldn't include exact versions here, because you would have to update the documentation with every upgrade. This would either make this document out-of-sync easily or require more effort for upgrades (we want neither). Instead you could refer to [podium-gateway/package.json](https://github.com/thehyve/podium/blob/dev/podium-gateway/package.json) for up-to-date dependency information.

Reply:
I agree, AngularJS version does not really belong here. NodeJS, on the other hand, can not be reliably derived from the package.json, so I'm going to keep it (and update it along with the app).